connection.go

```diff
@@ -190,8 +201,11 @@ func (ws *websocket) Read() (messageType int, msg []byte, err error) {
 	}
 
 	// ensure that we have the connection during the whole read operation
-	ws.mux.RLock()
-	defer ws.mux.RUnlock()
+	ws.read.Lock()
+
+	defer func() {
+		ws.read.Unlock()
```

Comment:
why not just
`defer ws.read.Unlock()`

Reply:
will remove, had some logging in there before